src/lib.rs

```diff
@@ -1,11 +1,38 @@
-use bitcoin::ScriptBuf as RustBitcoinScriptBuf;
+use bitcoin::Amount as BitcoinAmount;
+use bitcoin::ScriptBuf as BitcoinScriptBuf;
+
+use error::ParseAmountError;
+
+pub mod error;
+pub use bitcoin::amount::ParseAmountError as BitcoinParseAmountError;
+pub use bitcoin::Network;
+
+macro_rules! impl_from_core_type {
+    ($inner:ident, $outer:ident) => {
+        impl From<$inner> for $outer {
+            fn from(inner: $inner) -> Self {
+                $outer(inner)
+            }
+        }
+    };
+}
+
+macro_rules! impl_from_outer_type {
```

Comment:
Same comment as above basically.

Reply:
I also think it's more intuitive to reverse the order of the arguments here, so that the call site looks like this:

```rust
impl_from_ffi_type!(Amount, BitcoinAmount);
```

Reply:
Do you think these make sense to move to a "`macros.rs`" or "`utils.rs`"

Reply:
Yeah I would have them in a macro.rs file.

Reply:
Can't do macro singular cause that's a keyword. Went with `macros`, going to force push in a sec